cross-project-tests/tools/llvm-objdump/ARM/plt.c

```diff
@@ -0,0 +1,243 @@
+// Test PLT section parsing on arm.
+
+// RUN: %clang --target=armv6a-none-linux-gnueabi -fuse-ld=lld \
+// RUN:   -nostdlib -nostdinc -shared %s -o %t1
+// RUN: llvm-objdump --no-show-raw-insn --no-print-imm-hex \
+// RUN:   -d %t1 | FileCheck %s --check-prefixes=CHECK,LE
+
+// Test PLT section parsing on armeb.
+
+// RUN: %clang --target=armv6aeb-none-linux-gnueabi -fuse-ld=lld \
+// RUN:   -nostdlib -nostdinc -shared %s -o %t2
+// RUN: llvm-objdump --no-show-raw-insn --no-print-imm-hex \
+// RUN:   -d %t2 | FileCheck %s --check-prefixes=CHECK,BE
+// RUN: obj2yaml %t2 | FileCheck %s --check-prefixes=NOBE8
+
+// Test PLT section parsing on armeb with be8.
+
+// RUN: %clang --target=armv7aeb-none-linux-gnueabi -fuse-ld=lld \
+// RUN:   -nostdlib -nostdinc -shared %s -o %t3
+// RUN: llvm-objdump --no-show-raw-insn --no-print-imm-hex \
+// RUN:   -d %t3 | FileCheck %s --check-prefixes=CHECK,BE
+// RUN: obj2yaml %t3 | FileCheck %s --check-prefixes=BE8
+
+// LE: file format elf32-littlearm
+// BE: file format elf32-bigarm
+// NOBE8-NOT: EF_ARM_BE8
+// BE8: EF_ARM_BE8
+
+// CHECK: Disassembly of section .text:
+// CHECK-EMPTY:
+// CHECK-NEXT:    <_start>:
+// CHECK-NEXT:      push	{r11, lr}
+// CHECK-NEXT:      mov	r11, sp
+// CHECK-NEXT:      bl	{{.*}} <func1@plt>
+// CHECK-NEXT:      bl	{{.*}} <func2@plt>
+// CHECK-NEXT:      bl	{{.*}} <func3@plt>
+
+// CHECK: Disassembly of section .plt:
+// CHECK:      <func1@plt>:
+// CHECK-NEXT:      add	r12, pc, #0, #12
+// CHECK-NEXT:      add	r12, r12, #32, #20
+// CHECK-NEXT:      ldr	pc, [r12, #132]!
+// CHECK-NEXT:      .word	0xd4d4d4d4
+// CHECK-EMPTY:
+// CHECK-NEXT: <func2@plt>:
+// CHECK-NEXT:      add	r12, pc, #0, #12
+// CHECK-NEXT:      add	r12, r12, #32, #20
+// CHECK-NEXT:      ldr	pc, [r12, #120]!
+// CHECK-NEXT:      .word	0xd4d4d4d4
+// CHECK-EMPTY:
+// CHECK-NEXT: <func3@plt>:
+// CHECK-NEXT:      add	r12, pc, #0, #12
+// CHECK-NEXT:      add	r12, r12, #32, #20
+// CHECK-NEXT:      ldr	pc, [r12, #108]!
+// CHECK-NEXT:      .word	0xd4d4d4d4
+
+// Test PLT section parsing on thumb.
+
+// RUN: %clang --target=thumbv8.1m.main-none-linux-eabi \
+// RUN:   -nostdlib -nostdinc -c %s -o %t4.o
+// RUN: ld.lld --shared %t4.o -o %t4
+// RUN: llvm-objdump --no-show-raw-insn --no-print-imm-hex \
+// RUN:   -d %t4 | FileCheck %s --check-prefixes=THUMB,LE
+
+// Test PLT section parsing on thumbeb.
+
+// RUN: %clang --target=thumbebv8.1m.main-none-linux-eabi \
+// RUN:   -nostdlib -nostdinc -c %s -o %t5.o
+// RUN: ld.lld --shared %t5.o -o %t5
+// RUN: llvm-objdump --no-show-raw-insn --no-print-imm-hex \
+// RUN:   -d %t5 | FileCheck %s --check-prefixes=THUMB,BE
+// RUN: obj2yaml %t5 | FileCheck %s --check-prefixes=NOBE8
+
+// Test PLT section parsing on thumbeb with be8.
+
+// RUN: %clang --target=thumbebv8.1m.main-none-linux-eabi \
+// RUN:   -nostdlib -nostdinc -c %s -o %t6.o
```

Comment:
`-c` suppresses the link phase. `-nostdlib` should be removed, as it only affects linking.